taf/updater/updater_pipeline.py

```diff
@@ -802,23 +802,22 @@ def determine_start_commits(self):
                         last_validated_commit = last_validated_repository_commits_data[
                             "commit"
                         ]
-
+                        branch_exists = repository.branch_exists(
+                            current_branch, include_remotes=False
+                        )
+                        if not branch_exists:
+                            is_initial_state_in_sync = False
+                            break
                         if repository.is_bare_repository:
-                            # fetches the top commit of the remote branch (origin/{current_branch})
-                            remote_branch = f"origin/{current_branch}"
-                            top_commit_of_branch = repository.top_commit_of_branch(
-                                remote_branch
-                            )
+                            try:
+                                top_commit_of_branch = repository.top_commit_of_remote_branch(
```

Comment:
All of the checks that we have for regular repos should still be executed even if this is a bare repo. The only difference is that, since after being clone from disk to temp local branches do not seem to get created, checking if the branch exists and getting the top commit will need to done differently. Once we've checked if the branch exists and found the top commit, we still need:
```
    if top_commit_of_branch != last_validated_commit:
                            # check if top commit is newer (which is fine, it will be validated)
                            # or older, meaning that the authentication repository contains
                            # additional commits, so it would be necessary to find older auth repo
                            # commit and start the validation from there
                            if (
                                current_branch
                                not in repository.branches_containing_commit(
                                    last_validated_commit
                                )
                            ):
                                is_initial_state_in_sync = False
                                break
```
I'm worried that `branches_containing_commit` will not work if we don't have the local branches.

I've been looking at this and I can't remember why I even decided to iterate over temp repositories here. We might be able to get rid of all of this code (since local branches should exist by iterating over local repositories here `self.state.users_target_repositories`. Let's try doing that, and if there are any other issues, I'll debug tomorrow. We can keep the new git method once it's cleaned up in any case, it might be useful